app/models/mentor_request.rb

```diff
@@ -0,0 +1,32 @@
+class MentorRequest < ActiveRecord::Base
+  belongs_to :student, class_name: "User"
+  validates_presence_of :student
+
+  after_create :light_up
+
+  def light_up
+    Light.on
+  end
+    
+
+  def self.open_by_user(user)
+    req = self.where(student: user, open: true)
+  end
+
+  def self.close_all_by_user(user)
+    self.open_by_user(user).each{|req| req.close}
+  end
```

Comment:
Possibly you'd need to remember if there were any open that weren't yours and keep the light on if that was the case. Ideally there should be only one result of this query anyway I just made it close them all to be certain the db stayed clean.